modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/TypeScriptAngular2ClientCodegen.java

```diff
@@ -55,6 +57,74 @@ protected void addAdditionPropertiesToCodeGenModel(CodegenModel codegenModel, Mo
         addImport(codegenModel, codegenModel.additionalPropertiesType);
     }
 
+
+    /**
+     * Override of superclass method used to create a `vendorExtensions` entry
+     * on each model property for the code snippet that is used in the
+     * XyzModelSerializer.ts classes to deserialize the value.
+     *
+     * For example, this is used to deserialize date strings coming from a server
+     * into real JavaScript Date objects using the `new Date(dateStr)` constructor
+     * in the `serializer.mustache` template.
+     *
+     * @param objs
+     */
+    @Override
+    public Map<String, Object> postProcessAllModels(Map<String, Object> objs) {
+        objs = super.postProcessAllModels(objs);
+
+        List<CodegenModel> models = extractModels(objs);
+        for (CodegenModel model : models) {
+            for (CodegenProperty prop : model.vars) {
+                String snippet = createDeserializeSnippet(prop);
+
+                prop.vendorExtensions.put("deserializeSnippet", snippet);
```

Comment:
@gregjacobs using `vendorExtensions` is one way to add custom properties but I'm not sure if it's the best way for your situation/requirement.

Personally I would suggest one generic serializer class to handle serialization/deserialization for all model like what we've done for other languages: https://github.com/swagger-api/swagger-codegen/pull/4267#issuecomment-263163943